provisioning/service/src/DeviceRegistrationNotFoundException.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Common;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    using System;
+    using System.Runtime.Serialization;
+    using Microsoft.Azure.Devices.Common.Exceptions;
+
+#if !WINDOWS_UWP
+    [Serializable]
+#endif
+    public sealed class DeviceRegistrationNotFoundException : IotHubException
```

Comment:
Maybe we should create a new class for ProvisioningException?

---
In reply to: [141717282](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141717282) [](ancestors = 141717282)

Reply:
Since it is already in Provisioing.Service namespace.  I have removed "Device" to make it clearer.

---
In reply to: [141717282](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141717282) [](ancestors = 141717282)